### PR TITLE
Added --json-attributes option 

### PR DESCRIPTION
**Pull request by Vasu1105**
[In-progress] w.r.t https://github.com/chef-partners/azure-chef-extension/issues/164
The content to be added to Changelog is as follows:
* Added --json-attributes for ASM and ARM. User can set custom json attributes using this option which will get set in first_boot.json
Signed-off-by: Vasu1105 <vasundhara.jagdale@gmail.com>

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Vasu1105__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vmClient._js [hunk 1]
```diff
@@ -1138,6 +1138,16 @@ __.extend(VMClient.prototype, {
 
     pubConfig['client_rb'] = fs.readFileSync(options.clientConfig).toString();
     pubConfig['runlist'] = options.runList;
+
+    if (options.jsonAttributes) {
+      try {
+        pubConfig['custom_json_attr'] = JSON.parse(options.jsonAttributes);
+      }catch (er) {
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmClient._js]: } (missing space) catch

File: lib/commands/asm/vm/vmclient.js [hunk 2]
```diff
@@ -3162,6 +3162,15 @@ _.extend(VMClient.prototype, {
               config['client_rb'] = data.toString();
               config['runlist'] = options.runList;
               config['bootstrap_version'] = options.bootstrapVersion;
+              if (options.jsonAttributes) {
+                try {
+                  pubConfig['custom_json_attr'] = JSON.parse(options.jsonAttributes);
+                }catch (er) {
```
Comment (amarzavery) [inline lib/commands/asm/vm/vmclient.js]: same as above

File: lib/commands/arm/vm/vm._js [hunk 3]
```diff
@@ -633,6 +633,7 @@ exports.init = function (cli) {
               'In the format of \'name=value\'. Name is required and value is optional. For example, -t tag1=value1;tag2'))
       .option('-q, --quiet', $('quiet mode, do not ask for uninstall confirmation'))
       .option('-s, --subscription <id>', $('the subscription id'))
+      .option('--json-attributes <json-attributes>', $('A JSON string to be added to the first run of chef-client. Ex: --json-attributes \'{"foo" : "bar"}\''))
```
Comment (amarzavery) [inline lib/commands/arm/vm/vm._js]: @Vasu1105 - Can you add/update a test for this scenario?

Comment (Vasu1105) [inline lib/commands/arm/vm/vm._js]: @amarzavery yes will be adding those.

File: lib/commands/arm/vm/vmClient._js (hunk 1 above)
Comment (amarzavery) [inline lib/commands/arm/vm/vmClient._js]: @Vasu1105 - Please address these comments as well.

File: lib/commands/arm/vm/vm._js (hunk 3 above)
Comment (amarzavery) [inline lib/commands/arm/vm/vm._js]: var cmd = util.format('vm extension set-chef %s **-V** %s -c %s -O %s --json-attributes %s --json',

what is the **-V** used for? I dont see that option in the command definition for set-chef?

Comment (Vasu1105) [inline lib/commands/arm/vm/vm._js]: -V is for extension's version number.  This option is available for ASM mode

Comment (Vasu1105) [inline lib/commands/arm/vm/vm._js]: I have set following environment  variable to run in it record mode

set NOCK_OFF=
set AZURE_NOCK_RECORD=true

Comment (Vasu1105) [inline lib/commands/arm/vm/vm._js]: This is the output log
`
'2016-10-06T14:50:11.624Z  --Start--    cli vm extension: Set Chef extensions with custom json attributes: '
'node cli.js vm extension set-chef clitestvm8607 -V 1210.* -c test/data/set-chef-extension-client-config.rb -O test/data/set-chef-extension-validation.pem --json-attributes {"chef_node_name":"mynode"} --json'

Thu Oct 06 2016 20:20:20 GMT+0530 (IST):
'No VM with name "clitestvm8607" found'

Thu Oct 06 2016 20:20:20 GMT+0530 (IST):
{ [AssertionError: expected 1 to be 0]
  name: 'AssertionError',
  actual: 1,
  expected: 0,
  operator: 'to be',
  message: 'expected 1 to be 0',
  showDiff: true,
  generatedMessage: true }
'AssertionError: expected 1 to be 0\n    at Assertion.prop.(anonymous function) (/home/clogeny/workspace/azure-xplat-cli/node_modules/should/lib/should.js:61:14)\n    at /home/clogeny/workspace/azure-xplat-cli/test/commands/cli.vm.extension_set-chef.js:112:36\n    at /home/clogeny/workspace/azure-xplat-cli/test/util/util.js:206:7\n    at /home/clogeny/workspace/azure-xplat-cli/test/framework/cli-test.js:204:7\n    at /home/clogeny/workspace/azure-xplat-cli/test/framework/cli-executor.js:56:14\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/underscore/underscore.js:683:19\n    at process.<anonymous> (/home/clogeny/workspace/azure-xplat-cli/test/framework/cli-executor.js:68:7)\n    at Function.invoke (/home/clogeny/workspace/azure-xplat-cli/node_modules/sinon/lib/sinon/spy.js:197:51)\n    at process.proxy (/home/clogeny/workspace/azure-xplat-cli/node_modules/sinon/lib/sinon/spy.js:89:54)\n    at AzureCli.cli.__proto__.exitProcess (/home/clogeny/workspace/azure-xplat-cli/test/framework/cli-executor.js:74:63)\n    at AzureCli._.extend.exit (/home/clogeny/workspace/azure-xplat-cli/lib/cli.js:162:10)\n    at /home/clogeny/workspace/azure-xplat-cli/lib/cli.js:275:18\n    at Object.exports.onError (/home/clogeny/workspace/azure-xplat-cli/lib/util/telemetry.js:252:5)\n    at callback (/home/clogeny/workspace/azure-xplat-cli/lib/cli.js:265:21)\n    at /home/clogeny/workspace/azure-xplat-cli/lib/commands/asm/vm/vmclient.js:3063:16\n    at /home/clogeny/workspace/azure-xplat-cli/lib/commands/asm/vm/vmclient.js:2380:16\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/async/lib/async.js:52:16\n    at done (/home/clogeny/workspace/azure-xplat-cli/node_modules/async/lib/async.js:248:21)\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/async/lib/async.js:44:16\n    at /home/clogeny/workspace/azure-xplat-cli/lib/commands/asm/vm/vmclient.js:2376:11\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/azure-asm-compute/lib/computeManagementClient.js:8368:14\n    at /home/clogeny/workspace/azure-xplat-cli/lib/util/logging.js:339:7\n    at handleRedirect (/home/clogeny/workspace/azure-xplat-cli/lib/util/utils.js:564:9)\n    at /home/clogeny/workspace/azure-xplat-cli/lib/util/utils.js:587:7\n    at handleRedirect (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/lib/services/filters/redirectfilter.js:36:9)\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/lib/services/filters/errorhandlingfilter.js:49:7\n    at Request._callback (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/lib/http/request-pipeline.js:109:14)\n    at Request.self.callback (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/node_modules/request/request.js:187:22)\n    at Request.emit (events.js:98:17)\n    at Request.<anonymous> (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/node_modules/request/request.js:1044:10)\n    at Request.emit (events.js:117:20)\n    at IncomingMessage.<anonymous> (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/node_modules/request/request.js:965:12)\n    at IncomingMessage.emit (events.js:117:20)\n    at _stream_readable.js:944:16\n    at process._tickCallback (node.js:448:13)\n'
'2016-10-06T14:50:20.010Z  --End--    1)  cli vm extension: Set Chef extensions with custom json attributes'

'node cli.js vm delete clitestvm8607 -b -q --json'

'0 passing (47s)'
'1 failing'

'1) cli vm extension: Set Chef extensions with custom json attributes'

'Uncaught AssertionError: expected 1 to be 0'

'AssertionError: expected 1 to be 0\n    at Assertion.prop.(anonymous function) (/home/clogeny/workspace/azure-xplat-cli/node_modules/should/lib/should.js:61:14)\n    at /home/clogeny/workspace/azure-xplat-cli/test/commands/cli.vm.extension_set-chef.js:112:36\n    at /home/clogeny/workspace/azure-xplat-cli/test/util/util.js:206:7\n    at /home/clogeny/workspace/azure-xplat-cli/test/framework/cli-test.js:204:7\n    at /home/clogeny/workspace/azure-xplat-cli/test/framework/cli-executor.js:56:14\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/underscore/underscore.js:683:19\n    at process.<anonymous> (/home/clogeny/workspace/azure-xplat-cli/test/framework/cli-executor.js:68:7)\n    at Function.invoke (/home/clogeny/workspace/azure-xplat-cli/node_modules/sinon/lib/sinon/spy.js:197:51)\n    at process.proxy (/home/clogeny/workspace/azure-xplat-cli/node_modules/sinon/lib/sinon/spy.js:89:54)\n    at AzureCli.cli.__proto__.exitProcess (/home/clogeny/workspace/azure-xplat-cli/test/framework/cli-executor.js:74:63)\n    at AzureCli._.extend.exit (/home/clogeny/workspace/azure-xplat-cli/lib/cli.js:162:10)\n    at /home/clogeny/workspace/azure-xplat-cli/lib/cli.js:275:18\n    at Object.exports.onError (/home/clogeny/workspace/azure-xplat-cli/lib/util/telemetry.js:252:5)\n    at callback (/home/clogeny/workspace/azure-xplat-cli/lib/cli.js:265:21)\n    at /home/clogeny/workspace/azure-xplat-cli/lib/commands/asm/vm/vmclient.js:3063:16\n    at /home/clogeny/workspace/azure-xplat-cli/lib/commands/asm/vm/vmclient.js:2380:16\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/async/lib/async.js:52:16\n    at done (/home/clogeny/workspace/azure-xplat-cli/node_modules/async/lib/async.js:248:21)\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/async/lib/async.js:44:16\n    at /home/clogeny/workspace/azure-xplat-cli/lib/commands/asm/vm/vmclient.js:2376:11\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/azure-asm-compute/lib/computeManagementClient.js:8368:14\n    at /home/clogeny/workspace/azure-xplat-cli/lib/util/logging.js:339:7\n    at handleRedirect (/home/clogeny/workspace/azure-xplat-cli/lib/util/utils.js:564:9)\n    at /home/clogeny/workspace/azure-xplat-cli/lib/util/utils.js:587:7\n    at handleRedirect (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/lib/services/filters/redirectfilter.js:36:9)\n    at /home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/lib/services/filters/errorhandlingfilter.js:49:7\n    at Request._callback (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/lib/http/request-pipeline.js:109:14)\n    at Request.self.callback (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/node_modules/request/request.js:187:22)\n    at Request.emit (events.js:98:17)\n    at Request.<anonymous> (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/node_modules/request/request.js:1044:10)\n    at Request.emit (events.js:117:20)\n    at IncomingMessage.<anonymous> (/home/clogeny/workspace/azure-xplat-cli/node_modules/azure-common/node_modules/request/request.js:965:12)\n    at IncomingMessage.emit (events.js:117:20)\n    at _stream_readable.js:944:16\n    at process._tickCallback (node.js:448:13)\n'


`

Comment (Vasu1105) [inline lib/commands/arm/vm/vm._js]: @amarzavery I have added recording please review

**Pull request merged**